bottomless/src/replicator.rs

```diff
@@ -753,14 +767,15 @@ impl Replicator {
             )));
         let snapshot_notifier = self.snapshot_notifier.clone();
         let compression = self.use_compression;
+        let db_path = PathBuf::from(self.db_path.clone());
         let handle = tokio::spawn(async move {
             tracing::trace!("Start snapshotting generation {}", generation);
             let start = Instant::now();
-            let body = match Self::maybe_compress_main_db_file(db_file, compression).await {
+            let body = match Self::maybe_compress_main_db_file(&db_path, compression).await {
                 Ok(file) => file,
                 Err(e) => {
                     tracing::error!(
-                        "Failed to compress db file (generation {}): {}",
+                        "Failed to compress db file (generation {}): {:?}",
```

Comment:
Also changed error printout to Debug format - hopefully it will be more informative than `IO error` we got so far.